tests/br_log_restore/run.sh

```diff
@@ -87,13 +90,22 @@ run_sql "insert into ${DB}_DDL2.t2 values (4, 'x');"
 
 end_ts=$(run_sql "show master status;" | grep Position | awk -F ':' '{print $2}' | xargs)
 
+
 # if we restore with ts range [start_ts, end_ts], then the below record won't be restored.
 run_sql "insert into ${DB}_DDL2.t2 values (5, 'x');"
 
-# sleep wait cdc log sync to storage
-# TODO find another way to check cdc log has synced
-# need wait more time for cdc log synced, because we add some ddl.
-sleep 80
+wait_time=0
+checkpoint_ts=$(run_cdc cli changefeed query -c simple-replication-task --pd=https://$PD_ADDR | jq '.status."checkpoint-ts"')
+while (( checkpoint_ts < end_ts )); do
+    echo "waiting for cdclog syncing... (checkpoint_ts = $checkpoint_ts; end_ts = $end_ts)"
+    if (( wait_time > 300 )); then
+        echo "cdc failed to sync after 300s, please check the CDC log."
+        exit 1
+    fi
+    checkpoint_ts=$(run_cdc cli changefeed query -c simple-replication-task --pd=https://$PD_ADDR | jq '.status."checkpoint-ts"')
```

Comment:
may be we should swap line 105 and line 106, after we wait for 5s and get a new checkpoint_ts, otherwise we will get a old checkpoint_ts

Reply:
Yep, done.